src/cfi.rs

```diff
@@ -4885,4 +4941,31 @@ mod tests {
         let mut iter = section.entries(&bases);
         assert_eq!(iter.next(), Err(Error::FuncRelativePointerInBadContext));
     }
+
+    #[test]
+    fn register_rule_map_eq() {
+        // Different order, but still equal.
+        let map1: RegisterRuleMap<LittleEndian> = [(0, RegisterRule::SameValue),
+                                                   (3, RegisterRule::Offset(1))].iter().collect();
+        let map2: RegisterRuleMap<LittleEndian> = [(3, RegisterRule::Offset(1)),
+                                                   (0, RegisterRule::SameValue)].iter().collect();
+        assert_eq!(map1, map2);
+        assert_eq!(map2, map1);
+
+        // Not equal.
+        let map3: RegisterRuleMap<LittleEndian> = [(0, RegisterRule::SameValue),
+                                                   (2, RegisterRule::Offset(1))].iter().collect();
+        let map4: RegisterRuleMap<LittleEndian> = [(3, RegisterRule::Offset(1)),
+                                                   (0, RegisterRule::SameValue)].iter().collect();
+        assert!(map3 != map4);
+        assert!(map4 != map3);
+
+        // One has undefined, other implicitly has undefined.
+        let mut map5 = RegisterRuleMap::<LittleEndian>::default();
+        map5.set(0, RegisterRule::SameValue);
+        map5.set(0, RegisterRule::Undefined);
```

Comment:
Good catch. I think it makes sense to discard the option and return `RegisterRule::Undefined` if we don't have a rule: that's what we end up doing by the time we get to a public API anyways.